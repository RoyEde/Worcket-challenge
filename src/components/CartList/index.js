import React from 'react';
import { arrayOf, bool, func, shape, string } from 'prop-types';

import CartItem from './components/CartItem';

const CartList = ({ items, loading, onDelete }) => {
  return (
    <div className="cartlist">
      <h3 className="cartlist__count">
        {loading
          ? 'Loading...'
          : items.length === 0
          ? 'List is empty!'
          : `Items: ${items.length}`}
      </h3>
      <ul>
        {items.map(({ item, id }) => (
          <CartItem key={id} item={item} loading={loading} onDelete={() => onDelete(id)} />
        ))}
      </ul>
    </div>
  );
};

CartList.propTypes = {
  items: arrayOf(
    shape({
      id: string.isRequired,
      item: string.isRequired
    })
  ).isRequired,
  loading: bool,
  onDelete: func.isRequired
};

export default CartList;
