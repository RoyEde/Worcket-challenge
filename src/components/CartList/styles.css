.cartlist {
  margin-top: 1rem;
  display: flex;
  flex-direction: column;
  max-width: 768px;
  width: 100%;
}

.cartlist__item {
  align-items: center;
  background-color: #eeeeee;
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  margin: 0;
  list-style-type: none;
  padding: 0.5rem 1rem;
}
