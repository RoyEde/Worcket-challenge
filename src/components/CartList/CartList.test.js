import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import CartList from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CartList items={[]} onDelete={() => {}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshots match', () => {
  const component = renderer.create(<CartList items={[]} onDelete={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.items = [{ id: '0', item: 'test'}];
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.loading = true;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.items = [{ id: '0', item: 'test' }, { id: '1', item: 'test 2' },];
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.loading = false;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
