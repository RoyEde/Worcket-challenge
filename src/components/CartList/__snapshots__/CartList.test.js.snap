// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match 1`] = `
<div
  className="cartlist"
>
  <h2>
    List is empty!
  </h2>
  <ul />
</div>
`;

exports[`snapshots match 2`] = `
<div
  className="cartlist"
>
  <h2>
    List is empty!
  </h2>
  <ul />
</div>
`;

exports[`snapshots match 3`] = `
<div
  className="cartlist"
>
  <h2>
    List is empty!
  </h2>
  <ul />
</div>
`;

exports[`snapshots match 4`] = `
<div
  className="cartlist"
>
  <h2>
    List is empty!
  </h2>
  <ul />
</div>
`;

exports[`snapshots match 5`] = `
<div
  className="cartlist"
>
  <h2>
    List is empty!
  </h2>
  <ul />
</div>
`;
